repository.go

```diff
@@ -453,3 +454,129 @@ func (r *Repository) Reference(name plumbing.ReferenceName, resolved bool) (
 func (r *Repository) References() (storer.ReferenceIter, error) {
 	return r.s.IterReferences()
 }
+
+// ResolveRevision resolves revision to corresponding hash
+func (r *Repository) ResolveRevision(rev plumbing.Revision) (*plumbing.Hash, error) {
+	p := revision.NewParserFromString(string(rev))
```

Comment:
I followed instruction from mcuadro :

> Place all the parser, tokenizer, etc in an internal package called revision at the root of the package. This package deals with strings, not with the type Revision

Something is not clear, can you describe more how you want to organize revision ?

I added this method to repository cause I don't see why you would have to give a repository to solve a revision to a third part method, they are related closely to repository, like references. It seems strange to me to call it somewhere else, or at least if you do so it's great to have something to resolve them directly from repository as a convenience.